sensor/kubernetes/localscanner/service_certificates_repository.go

```diff
@@ -198,6 +204,10 @@ func (r *serviceCertificatesRepoSecretsImpl) ensureServiceCertificate(ctx contex
 	return patchErr
 }
 
+func (r *serviceCertificatesRepoSecretsImpl) getKnownServiceTypes() map[storage.ServiceType]serviceCertSecretSpec {
+	return r.secrets
```

Comment:
Hmm. Isn't it the situation here that you are returning an internal data structure to the caller who can then freely mutate it and hence also modify the behaviour of the service certificate repository? I would suggest to not offer such a possibility, instead returning a copy of the things the user really needs.